### ISSUE TITLE
Document that drat cannot work with git-lfs due to GH limitations

### ISSUE THREAD

**Issue by sjmgarnier** [eddelbuettel/drat#52]
A package that I am developing is larger than the 100mb limit of GitHub repositories (it contains OpenCV sources). I therefore opted to use git-lfs with my drat repository in order to allow this package to be pushed to my drat repository. Unfortunately it seems that drat does not download the tar.gz of the package when called from R but rather the text pointer created by git-lfs and pointing toward the actual location of the tar.gz.

**Comment by eddelbuettel** [comment]
Thanks for reporting this, but I don't exactly have a way forward on this as I don't use git-lfs.  

Maybe you could try?

**Comment by sjmgarnier** [comment]
@eddelbuettel I decided to take a different route for the moment so I don't need this anymore. However I'll keep this problem in the back of my head for when I have a bit of free time. It shouldn't be too difficult. Essentially, it should just require detecting that the drat repo is a git-lfs repo (easily done by checking the .gitattributes file) and then figuring out the address of the actual file rather than the address of the pointer.

**Comment by eddelbuettel** [comment]
Closing this at is no longer an open issue.

**Comment by bhaskarvk** [comment]
drat exposes gh-pages (or other such http accessible location) as a cran repo to R, and packages are installed/updated using the regular `install.packages` / `update.packages` commands. 
Unfortunately gh-pages doesn't support automatic redirection to the lfs enabled file. See https://github.com/git-lfs/git-lfs/issues/791, https://github.com/git-lfs/git-lfs/issues/1342. If gh-pages were ever to support that, then this will work seamlessly. Until then it may make sense to document this limitation in the package author vignette, just so the authors are aware of this limitation.

**Comment by eddelbuettel** [comment]
A repository (for users) is just a network readable resource (and the `drat:::add()` stuff is just sugar -- you can always use the full http://... address).  For authors it is the same for writable.  We "just" call the functions from the (base R) package tools.

I don't have git-lfs access so I can't test / push.  But I suppose we could work on this together.

**Comment by eddelbuettel** [comment]
Yes, that does sound right, and sorry for my confusion.   

We could then look into s3, glacier, ... or any other storage option.  It won't have the handy Github website with it, but we can still make drat support it.  

I'll reopen this and change the Subject so that I remember to update the documentation.

**Comment by bhaskarvk** [comment]
FWIW s3 can be exposed over HTTP, so you can have a website to go along with your repo. The only issue is by default the s3 domain name is not as pretty as GH pages, but if one has access to a nameserver, then you can always add a CNAME

**Comment by bryanhanson** [comment]
I have not used drat, I built my repo manually, but ended up on this page with the problem that installing via `install.packages` doesn't work with git-lfs (one underlying question here).  After some exploration, I was able to make it work; see my comment [here](https://github.com/git-lfs/git-lfs/issues/1342) as to how I solved the access issue outside of drat.  If drat is to work with git-lfs, I think there would have to be a function or option to initiate lfs tracking so Github knows which files are to be handled that way.  There is also a sequence of steps issue: you must turn on tracking _before_ you add the large file to the repo, the add, commit and push.  Otherwise the large files are not properly handled.
